src/core/distributions/phylogenetics/tree/birthdeath/GeneralizedLineageHeterogeneousBirthDeathSamplingProcess.cpp

```diff
@@ -0,0 +1,2042 @@
+#include <string>
+
+#include "RandomNumberFactory.h"
+#include "RandomNumberGenerator.h"
+#include "AbstractHomologousDiscreteCharacterData.h"
+#include "GeneralizedLineageHeterogeneousBirthDeathSamplingProcess.h"
+#include "RlAbstractHomologousDiscreteCharacterData.h"
+#include "RlString.h"
+#include "HomologousDiscreteCharacterData.h"
+#include "DiscreteCharacterState.h"
+#include "DiscreteTaxonData.h"
+#include "NaturalNumbersState.h"
+#include "TopologyNode.h"
+#include "Taxon.h"
+#include "Tree.h"
+#include "RlUserInterface.h"
+
+
+using namespace RevBayesCore;
+
+/**
+ * Constructor.
+*/
+GeneralizedLineageHeterogeneousBirthDeathSamplingProcess::GeneralizedLineageHeterogeneousBirthDeathSamplingProcess(
+	const std::vector<Taxon>&                                       taxa_,
+	const TypedDagNode<double>*                                     age_,
+	const std::string&                                              condition_type_,
+	const TypedDagNode<Simplex >*                                   root_frequency_,
+	const size_t                                                    num_states_,
+	bool                                                            use_origin_,
+	bool                                                            zero_indexed_,
+	size_t                                                          n_proc_
+) : TypedDistribution<Tree>( new TreeDiscreteCharacterData() ),
+	current_ln_prob(0.0),
```

Comment:
Perhaps consider initializing all default values in the class declaration.
Perhaps use `nullptr` instead of NULL.

Reply:
Done.